### DISCUSSION TITLE
size of stbi__zdist_extra in stb_image.h

### DISCUSSION THREAD

**Discussion by rds1983** [nothings/stb#1267] [Q&A] [answered]
I work upon porting stb libs to other languages and have a small question regarding this line:
```
static const int stbi__zdist_extra[32] =
{ 0,0,0,0,1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10,10,11,11,12,12,13,13};
```

There are only 30 values in the initializer list, while the array's declared size is 32.
So which size is correct: 30 or 32? 
If it is 32, then what are the last two values? zeroes?

**Comment by nothings** [comment, marked as answer]
The definition of DEFLATE defines 30 entries in that table. It is unclear if there is a reason the array is declared as 32 long. The C code will initialize the extra values to 0.

**Reply by rds1983** [reply]
Thanks!
